analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/CreatingHashAlgorithmsBase.cs

```diff
@@ -28,23 +28,48 @@ public abstract class CreatingHashAlgorithmsBase<TSyntaxKind> : SonarDiagnosticA
         protected const string DiagnosticId = "S4790";
         protected const string MessageFormat = "Make sure this weak hash algorithm is not used in a sensitive context here.";
 
-        protected ObjectCreationTracker<TSyntaxKind> ObjectCreationTracker { get; set; }
+        private const string CreateMethodName = "Create";
+
+        private readonly KnownType[] algorithmTypes =
+        {
+            KnownType.System_Security_Cryptography_DSA,
+            KnownType.System_Security_Cryptography_HMACMD5,
+            KnownType.System_Security_Cryptography_HMACRIPEMD160,
+            KnownType.System_Security_Cryptography_HMACSHA1,
+            KnownType.System_Security_Cryptography_MD5,
+            KnownType.System_Security_Cryptography_RIPEMD160,
+            KnownType.System_Security_Cryptography_SHA1
+        };
 
-        protected InvocationTracker<TSyntaxKind> InvocationTracker { get; set;  }
+        private readonly string[] unsafeAlgorithms = { "SHA1", "MD5", "DSA", "HMACMD5", "HMACRIPEMD160", "RIPEMD160", "RIPEMD160Managed" };
```

Comment:
"HMACSHA1" is missing in this list and UTs.

Reply:
Yes, it wasn't in the initial implementation of S2070. I will add it.